File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        if ( !root )
            return {};
        
        std::vector<int> result;
        std::queue<TreeNode *> q;
        q.push( root );
        
        while( !q.empty() ) {
            int lSize = q.size();
            int rightMost = 0;
            
            for(int i = 0; i < lSize; i++) {
                TreeNode *tmp = q.front();
                q.pop();
                rightMost = tmp->val;
                if ( tmp->left )
                    q.push(tmp->left);
                if ( tmp->right )
                    q.push(tmp->right);
            }
            result.push_back(rightMost);
        }
        return result;    
    }
};

File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def rightSideView(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        
        if root is None:
            return []
    
        result = []

        q = collections.deque([root])
        while len(q) > 0:
            numNode = len(q)
            lastElem = None
            for _ in range(numNode):
                node = q.popleft()
                lastElem = node.val
                if node.left is not None:
                    q.append(node.left)
                if node.right is not None:
                    q.append(node.right)
            result.append(lastElem)
        
        return result